core/market/decentralized/src/matcher.rs

```diff
@@ -148,26 +145,27 @@ impl Matcher {
     ) -> Result<(), MatcherError> {
         Ok(self.store.remove_demand(subscription_id).await?)
     }
-
-    pub fn emit_proposal(&self, proposal: DraftProposal) -> Result<(), SendError<DraftProposal>> {
-        self.proposal_emitter.send(proposal)
-    }
 }
 
 pub(crate) async fn on_offer_received(
-    store: SubscriptionStore,
+    resolver: Resolver,
     _caller: String,
     msg: OfferReceived,
 ) -> Result<Propagate, ()> {
     // We shouldn't propagate Offer, if we already have it in our database.
     // Note that when we broadcast our Offer, it will reach us too, so it concerns
     // not only Offers from other nodes.
 
-    store
-        .store_offer(msg.offer)
+    let subscription = Subscription::from(&msg.offer);
+    resolver
+        .store
+        .save_offer(msg.offer)
         .await
         .map(|propagate| match propagate {
```

Comment:
This should not be named `propagate`, but probably `inserted` or something like this. Save_offer doesn't return info, if we should propagate anything

Reply:
true